src/pstore/pstore.c

```diff
@@ -0,0 +1,735 @@
+
+                snprintf(pathname, sizeof(pathname), "%s/%s", sourcepath, de->d_name);
+                if (0 == lstat(pathname, &statbuf))
+                {
+                    if (NULL != (dc = (dirent_ct *)malloc(sizeof(dirent_ct))))
+                    {
+                        dc->de = *de;
+                        dc->statbuf = statbuf;
+                        dc->is_binary = 1;
+                        append_dirent_to_list(list, dc);
+
+                        /* Now read contents of pstore file */
+                        dc->contents = (char *)malloc(dc->statbuf.st_size);
+                        if (NULL != dc->contents)
+                        {
+                            int ifd;
+                            ssize_t rrc;
+
+                            ifd = open(pathname, O_RDONLY);
+                            if (ifd < 0)
+                            {
+                                log_error("open(%s): %s\n", pathname, strerror(errno));
+                            }
+                            else
+                            {
+                                rrc = read(ifd, dc->contents, dc->statbuf.st_size);
+                                if (rrc != dc->statbuf.st_size)
+                                {
+                                    log_error("read(%s, %ld) only returned %ld bytes\n", pathname, dc->statbuf.st_size, rrc);
+// FIX??? do we keep partial bytes or throw all away?
+                                }
+                            }
+                        }
+                        else
+                        {
+                            log_error("malloc(%s, %ld): failed\n", pathname, dc->statbuf.st_size);
+                        }
+                    }
+                    else
+                    {
+                        log_error("malloc() failed: out of memory!\n");
+                    }
+                }
+                else
+                {
+                    log_error("lstat(%s): %s\n", pathname, strerror(errno));
+                    rc = errno;
+                    break;
+                }
+            }
+            else
+            {
+                if (errno)
+                {
+                    log_error("readdir(): %s\n", strerror(errno));
+                    rc = errno;
+                    break;
+                }
+                /* else end of dir */
+            }
+        } while (NULL != de);
+    }
+    return rc;
+}
+
+static int
+run (int argc, char *argv[])
+{
+    unsigned i;
+    int process_pstore = 0;
+    int rc = 0;
+    dirent_ct *fileList = NULL;
+
+    log_open();
+
+    /* Ignore all parse errors */
+    (void) parse_config();
+
+    log_debug("Selected storage '%s'.", pstore_storage_to_string(arg_storage));
+    log_debug("Selected SourceDir '%s'.", arg_sourcedir);
+    log_debug("Selected ArchiveDir '%s'.", arg_archivedir);
+    log_debug("Selected AllowUnlink '%d'.", arg_allowunlink);
+
+    if (PSTORE_STORAGE_NONE == arg_storage)
```

Comment:
Please don't do yoda comparisons. There is really no benefit whatsoever. It made some sense back in the day when compilers were stupid, but nowadays, if you were to mistype == as =, the compiler will emit a nice and precise warning.

Reply:
ok, ive changed the yoda comparisons